src/backend/downloadmanager/utils.ts

```diff
@@ -0,0 +1,108 @@
+import { BrowserWindow } from 'electron'
+import { logError, logInfo, LogPrefix, logWarning } from '../logger/logger'
+import { getGame, isEpicServiceOffline, notify } from '../utils'
+import { InstallParams } from 'common/types'
+import i18next from 'i18next'
+import { showErrorBoxModalAuto } from '../dialog/dialog'
+import { isOnline } from '../online_monitor'
+
+async function installQueueElement(
+  mainWindow: BrowserWindow,
+  params: InstallParams
+) {
+  const {
+    appName,
+    path,
+    installDlcs,
+    sdlList = [],
+    runner,
+    installLanguage,
+    platformToInstall
+  } = params
+  const game = getGame(appName, runner)
+  const { title } = game.getGameInfo()
+
+  if (!isOnline()) {
+    logWarning(`App offline, skipping install for game '${title}'.`, {
+      prefix: LogPrefix.Backend
+    })
+    return { status: 'error' }
+  }
+
+  const epicOffline = await isEpicServiceOffline()
+  if (epicOffline && runner === 'legendary') {
+    showErrorBoxModalAuto({
+      title: i18next.t('box.warning.title', 'Warning'),
+      error: i18next.t(
+        'box.warning.epic.install',
+        'Epic Servers are having major outage right now, the game cannot be installed!'
+      )
+    })
+    return { status: 'error' }
+  }
+
+  mainWindow.webContents.send('setGameStatus', {
```

Comment:
I think we should get the mainWindow right here and check if present, check my previous comment on the function that calls this, I'm worried that this will not always reference the right thing after heroic is closed to the tray